core/src/main/java/org/mobilitydata/gtfsvalidator/input/GtfsInput.java

```diff
@@ -56,11 +63,100 @@ public static GtfsInput createFromPath(Path path) throws IOException {
       // Read from a local ZIP file.
-      return new GtfsZipFileInput(new ZipFile(path.toFile()));
+      zipFile = new ZipFile(path.toFile());
+      if (containsSubfolderWithGtfsFile(path)) {
+        noticeContainer.addValidationNotice(
+            new InvalidInputFilesInSubfolderNotice(invalidInputMessage));
+      }
+      return new GtfsZipFileInput(zipFile);
     }
     // Load a remote ZIP file to memory.
-    return new GtfsZipFileInput(
-        new ZipFile(new SeekableInMemoryByteChannel(Files.readAllBytes(path))));
+    zipFile = new ZipFile(new SeekableInMemoryByteChannel(Files.readAllBytes(path)));
+    if (containsSubfolderWithGtfsFile(path)) {
+      noticeContainer.addValidationNotice(
+          new InvalidInputFilesInSubfolderNotice(invalidInputMessage));
+    }
+    return new GtfsZipFileInput(zipFile);
+  }
+
+  /**
+   * Check if a zip file contains a subfolder with GTFS files
+   *
+   * @param path
+   * @return
+   */
+  public static boolean containsSubfolderWithGtfsFile(Path path) {
+    boolean containsGtfsFileInSubfolder = false;
+    boolean containsSubfolder = false;
+    String subfolder = null;
+    try (ZipInputStream zipInputStream =
+        new ZipInputStream(new BufferedInputStream(new FileInputStream(path.toFile())))) {
+      ZipEntry entry;
+      while ((entry = zipInputStream.getNextEntry()) != null) {
+        String entryName = entry.getName();
+        if (entry.isDirectory()) {
+          containsSubfolder = true;
+          subfolder = entry.getName();
+        }
+        if (containsSubfolder && entryName.contains(subfolder) && entryName.endsWith(".txt")) {
+          String[] files = entryName.split("/");
+          String lastElement = files[files.length - 1];
+          if (containsGtfsFile(lastElement)) {
+            containsGtfsFileInSubfolder = true;
+          }
+        }
+      }
+    } catch (IOException ioException) {
+      logger.atSevere().withCause(ioException).log(
+          "GtfsInput containsSubfolderWithTxtFile throws IOException");
+    } finally {
+      return containsGtfsFileInSubfolder;
+    }
+  }
+
+  public static boolean containsSubfolderWithGtfsFile(URL url) {
+    boolean containsGtfsFileInSubfolder = false;
+    boolean containsSubfolder = false;
+    String subfolder = null;
+    try (ZipInputStream zipInputStream =
+        new ZipInputStream(new BufferedInputStream(url.openStream()))) {
+      ZipEntry entry;
+      while ((entry = zipInputStream.getNextEntry()) != null) {
+        String entryName = entry.getName();
+        if (entry.isDirectory()) {
+          containsSubfolder = true;
+          subfolder = entry.getName();
+        }
+        if (containsSubfolder && entryName.contains(subfolder) && entryName.endsWith(".txt")) {
+          String[] files = entryName.split("/");
+          String lastElement = files[files.length - 1];
+          if (containsGtfsFile(lastElement)) {
+            containsGtfsFileInSubfolder = true;
```

Comment:
that's right

Reply:
but, after this if statement, the program goes to the finally block and return the containsGtfsFileInSubfolder variable. If I removed containsGtfsFileInSubfolder variable, I still need to put a return false at the end. I prefer to keep it in this way.

Reply:
In that case I would suggest you get out of the while loop with a break immediately after setting containsGtfsFileInSubfolder to true. That way it will not continue looking at the zip file for something it already found.

Reply:
I'm on it now.

Reply:
Done